src/translator/html.cpp

```diff
@@ -336,48 +330,86 @@ AnnotatedText RestoreTarget(AnnotatedText const &in, std::vector<HTML::Taint> co
 
     size_t offset = 0;  // Size added by prepending HTML
     size_t whitespace_size = CountPrefixWhitespaces(token);
+    bool close_left = true;  // See RestoreSource's implementation
+
+    // First we scan through spans_ to catch up to the span assigned to this
+    // token. We're only interested in empty spans (empty and void elements)
+    for (auto span_it = prev_span + 1; span_it < *target_span_it; span_it++) {
+      // We're only interested in empty spans between the spans in target_span_it
+      if (span_it->size() != 0) continue;
+
+      DiffTags(prev_span->tags, span_it->tags, opening, closing);
+
+      for (auto cit = closing.crbegin(); cit != closing.crend(); ++cit) {
+        std::string close_tag = format("</{}>", (*cit)->name);
```

Comment:
`fmt::format` is available via spdlog, although a bit old. This is what is used by marian's `LOG(fmtstring, Args...)`. Are there speed implications for maintaining a new one, if not let's reuse that one? I've tried getting in the updated one, but ends up violating ODR without a trickle down.

Reply:
It's there mainly for readability because string concatenations are so verbose in C++.

It's a very simple implementation that in itself should be pretty fast. However, it uses std::ostringstream under the hood which is maybe quite the hog. If profiling points that out, we should probably replace these calls with just some raw string operations, and the usage around the exceptions with `fmt::format`.

Reply:
Resolving. Is not strictly related to this PR anyway, but just poked a query given we rushed the first PR.